### PR TITLE
(#581) Add tests for the sitemap.

### LINKED ISSUE #581: Enabler: Create tests for /sitemap.xml
# Verify the sitemap is present.

*Description of what is to be accomplished.*
*Examples:*
* *Initial Setup of Dev Tier*
* *Create Application Framework*
* *Define Unit Test Approach*
* *Research UX Patterns for Search*

> **ESTIMATE** TBD

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* References: *URL to web site with examples*
* Web Page: *Here goes a URL to an existing web page*

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Sub-Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…
* More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by blairlearn**
Closes #581

### CODE REVIEW

File: cypress/e2e/Sitemap/sitemap.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        Then the sitemap includes the path "<path>"
+
+        Examples:
+            ## It is expected that all site map entries point to www.cancer.gov as this is the URL we want indexed if a search engine
+            ## indexes a lower-tier server.
+            ## Commented out entries are for types which are excluded from sitemaps
+            | path                                                                              | type                           |
+            | https://www.cancer.gov/                                                           | Home and Landing page          |
+            | https://www.cancer.gov/espanol                                                    |                                |
+            #| https://www.cancer.gov/search/results                                               | Application page              |
```
Comment (KateMashkinaNIH) [inline cypress/e2e/Sitemap/sitemap.feature:23]: are commented out entries intended to be enabled in the future? If not, then its better to remove them all together

File: cypress/e2e/common/sitemapSupport.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+        expect(response).to.have.property('headers');
+        const actualType = response.headers['content-type'];
+
+        expect(actualType).to.equal(expectedType);
+    });
+});
+
+
+Then("the sitemap has at least {int} urls", (minimumCount) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/common/sitemapSupport.js:20]: I would suggest to combine the min and max step into one - seeing how the code is duplicated, and only assertion portion is different 

Then('the sitemap has at least {int} and no more than {int} urls',(minimumCount, maximumCount)=>{

    cy.get("@sitemap_response").should((response) => {
        expect(response).to.have.property('body');

        const urls = Cypress.$(response.body)
            .find("loc")
            .toArray()
            .map((el) => el.innerText);

        expect(urls.length).to.be.at.least(minimumCount);
        expect(urls.length).to.be.lessThan(maximumCount)
    });
});

File: cypress/e2e/common/sitemapSupport.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            .toArray()
+            .map((el) => el.innerText);
+
+        expect(urls.length).to.be.lessThan(maximumCount);
+    });
+});
+
+
+Then("the sitemap has at least {int} alternate entries with language code {string}", (minimumCount, langcode) => {
+
```
Comment (KateMashkinaNIH) [inline cypress/e2e/common/sitemapSupport.js:51]: same here - combine the two

File: cypress/e2e/common/sitemapSupport.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        urls.forEach( entry => expect(entry).to.match(re));
+    });
+});
+
+Then('the sitemap includes the path {string}', (expectedPath) => {
+
+        cy.get("@sitemap_response").should((response) => {
+        expect(response).to.have.property('body');
+
+        const re = /https?:\/\/.*?(\/.*)/;
```
Comment (KateMashkinaNIH) [inline cypress/e2e/common/sitemapSupport.js:99]: this const is not used in the step - should be removed

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]
sitemap step definitions are pretty specific to sitemap.feature - create a sitemap folder ( named exactly like sitemap.feature file) under the Sitemap folder and move it there from common

File: cypress/e2e/common/sitemapSupport.js (hunk 2 above)
Comment (blairlearn) [reply cypress/e2e/common/sitemapSupport.js:20]: That's fair.

File: cypress/e2e/common/sitemapSupport.js (hunk 3 above)
Comment (blairlearn) [reply cypress/e2e/common/sitemapSupport.js:51]: Ditto

File: cypress/e2e/common/sitemapSupport.js (hunk 4 above)
Comment (blairlearn) [reply cypress/e2e/common/sitemapSupport.js:99]: Whoops! That's a leftover.

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**